app/connectors/Cache.scala

```diff
@@ -35,6 +35,9 @@ class MongoCacheConnector @Inject()(val repository: CacheMapRepository) extends
 
   def getEntry[A](cacheId: String, key: String)(implicit hc: HeaderCarrier, fmt: Format[A]): Future[Option[A]] =
     fetch(cacheId).map(_.flatMap(_.getEntry(key)))
+
+  def remove()(implicit hc: HeaderCarrier): Unit =
+    repository.remove()
```

Comment:
Remove method probably needs to be based on the `cacheId`.

Reply:
That's not true, http-caching-client retrieve the cache ID from header carrier

Reply:
`repository.remove()` return `Future[HttpResponse]` not unit.

```
def remove()(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[HttpResponse] =
    for {
      c      <- cacheId
      result <- delete(buildUri(defaultSource, c))
    } yield result
```

Can you change the return type?